azurerm/resource_arm_container_group.go

```diff
@@ -762,6 +920,73 @@ func expandContainerVolumes(input interface{}) (*[]containerinstance.VolumeMount
 	return &volumeMounts, &containerGroupVolumes
 }
 
+func expandContainerProbe(input interface{}) *containerinstance.ContainerProbe {
+	probe := containerinstance.ContainerProbe{}
+	probeRaw := input.([]interface{})
+
+	if len(probeRaw) == 0 {
+		return nil
+	}
+
+	for _, p := range probeRaw {
+		probeConfig := p.(map[string]interface{})
+
+		if initialDelaySeconds := probeConfig["inital_delay_seconds"].(int); initialDelaySeconds > 0 {
```

Comment:
_minor and not a blocker_ but you could use v here to simplify this a little:
```go
		if v := probeConfig["inital_delay_seconds"].(int); v > 0 {
			probe.InitialDelaySeconds = utils.Int32(int32(v))
		}
```

But totally optional change